### ISSUE TITLE
domainObjectRenamingRule has bug when replace prefix

### ISSUE THREAD

**Issue by itfsw** [mybatis/generator#359]
new config domainObjectRenamingRule has bug when replace prefix.
```xml
<domainObjectRenamingRule searchString="^T" replaceString="" />
```
when we use this rule to replace our tables,the class will not uppercase the first letter of table.

**Comment by itfsw** [comment]
in org.mybatis.generator.api.IntrospectedTable#getDomainObjectName should be
```java
/**
     * Gets the domain object name.
     *
     * @return the domain object name
     */
    public String getDomainObjectName() {
        if (stringHasValue(domainObjectName)) {
            return domainObjectName;
        }

        String finalDomainObjectName;
        if (stringHasValue(runtimeTableName)) {
            finalDomainObjectName =  getCamelCaseString(runtimeTableName, true);
        } else {
            finalDomainObjectName =  getCamelCaseString(introspectedTableName, true);
        }

        if (domainObjectRenamingRule != null) {
            Pattern pattern = Pattern.compile(domainObjectRenamingRule.getSearchString());
            String replaceString = domainObjectRenamingRule.getReplaceString();
            replaceString = replaceString == null ? "" : replaceString; //$NON-NLS-1$
            Matcher matcher = pattern.matcher(finalDomainObjectName);
            finalDomainObjectName = matcher.replaceAll(replaceString);
            // uppercase the first letter
            finalDomainObjectName = getCamelCaseString(finalDomainObjectName, true);
        }
        return finalDomainObjectName;
    }
```

**Comment by feixialong** [comment]
this function now just upper first letter ,but other letter also small letter.
the table name is t_risk_data  . i want it to RiskData, but when i using 
<domainObjectRenamingRule searchString="^T" replaceString="" />
it become to Riskdata

**Comment by jeffgbutler** [comment]
This is fixed in #382 - not released yet. Please try with the latest snapshot.

**Comment by tangyikai** [comment]
I've found this bug in 1.3.7. However, instead of using snapshot version, I've just replaced 1.3.7 version with 1.3.6 and everything went perfectly (because I haven't use any new feature in 1.3.7). 

If you are trying to solve this problem before the next version is released, I think the most efficient way is using a previous version.


```
//version 1.3.6 
if (domainObjectRenamingRule != null) {
            Pattern pattern = Pattern.compile(domainObjectRenamingRule.getSearchString());
            String replaceString = domainObjectRenamingRule.getReplaceString();
            replaceString = replaceString == null ? "" : replaceString; //$NON-NLS-1$
            Matcher matcher = pattern.matcher(finalDomainObjectName);
            finalDomainObjectName = matcher.replaceAll(replaceString);
        }
```
